src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -561,6 +563,9 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this.updateConfig();
 				this.setVisible(this._isVisible);
 			}
+			if (e.affectsConfiguration(TerminalSettingId.ShellIntegrationEnvironmentReporting)) {
+				this._shellLaunchConfig.shellIntegrationEnvironmentReporting = this._configurationService.getValue(TerminalSettingId.ShellIntegrationEnvironmentReporting);
```

Comment:
I don't think you need this since it's handled in the constructor? This decision cannot be undone when config changes after the terminal is created. The terminals must be re-created.

Reply:
good point yeah, you're right the terminal has to be re-created for config to be in effect. Resolved here  ---> https://github.com/microsoft/vscode/pull/240424/commits/c017f3cbd2f155453c9f29f0e9ebbcb03cecae73